polaris-react/src/components/DataTable/DataTable.scss

```diff
@@ -95,11 +95,6 @@
 .TableRow + .TableRow {
   .Cell {
     border-top: var(--p-border-width-1) solid var(--p-color-border-subdued);
-
-    #{$se23} & {
-      border-top: var(--p-border-width-1) solid
-        var(--p-color-border-faint-experimental);
-    }
```

Comment:
![image](https://github.com/Shopify/polaris/assets/612020/2e22eee2-84fe-4352-958e-bafbd6ce03e6)

Reply:
Looks like this came about from removing the `faint` border... Probably my bad, sorry!

Reply:
You're good! I should have rebased my DataTable branch before merging (after being off for 4 days). No worries, easy fix 🙌